### ISSUE TITLE
OpenBVE not loading routes on Mac

### ISSUE THREAD

**Issue by cat1554** [leezer3/OpenBVE#409]
### Description

The latest macOS version doesn't load any routes, & just kicks back to the menu.

### Reproduction

I don't have access to other computers, but this issue occurs with all routes on my MacBook.

## Logs

[openBVE Bug Report04_12_2019.zip](https://github.com/leezer3/OpenBVE/files/3924516/openBVE.Bug.Report04_12_2019.zip)

### Related information 

* MacOS 10.14.6
MacBook Pro (13-inch, Mid 2012)
2.5 GHz Intel Core i5
8 GB 1600 MHz DDR3
Intel HD Graphics 4000 1536 MB

* OpenBVE 1.7.1.0
* Keyboard

**Comment by leezer3** [comment]
This would appear to be a mostly Apple-centric issue :(

It's a little late for me to be testing / fixing this tonight, so will roll the site back to 1.7.0 for the minute on Mac.

TLDR:
Apple appear to maintain two different openGL contexts, one which supports GSL 3.0+ and one that supports below this.
Everybody else just supports the whole bunch, but no.....

This has also highlighted a separate issue whereby everything falls over nastily if we fail to create the default shader.

cc @s520

**Comment by leezer3** [comment]
The commit above is a partial fix, so that we no longer dump out when creating the default shader fails. I *think* this makes the loading screen work OK, but it'll then crash when it attempts to draw the background.....
Still todo:
* Fix background rendering so that the old code works again.
* Disable the new renderer on Mac (??)
* Try and figure out exactly what combinations are supported where...

**Comment by s520** [comment]
It seems necessary to use a core profile of 3.2 or higher on macOS.
https://www.qt.io/blog/2017/01/27/opengl-core-profile-context-support-qpainter
https://stackoverflow.com/questions/20264814/glsl-version-130-on-mac-os-x-causes-error

I think it is better to switch between GLSL versions when running on macOS.

**Comment by leezer3** [comment]
I've seen that and some similar stuff.

As I read it, we can either use 3.2 *or* 2.0 and below.
3.2 core context doesn't support immediate mode I think, which would mean the text drawing and IIRC some other stuff needs some work....

**Comment by leezer3** [comment]
This version will hopefully fall back gracefully from the error you're seeing:
https://vps.bvecornwall.co.uk/OpenBVE/Builds/OpenBVE-2019-12-09.dmg

The new renderer will not work (checkbox does nothing), and there may *still* be a glitch or two, but I think it should at least now load most things.

(I haven't got access to a real Mac at the minute, so I've had to improvise a little on the testing)

**Comment by leezer3** [comment]
@cat1554 https://vps.bvecornwall.co.uk/OpenBVE/Builds/OpenBVE-2020-01-01.dmg
I believe this build ought to work correctly with all features.

Are you able to test?
